prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,6 +204,56 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Get a sortable list of facilities for multiple deliveries
+     *
+     * @param request HTML request body.
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilitiesBulk")
+    fun getDeliveryFacilitiesBulk(
+        @HttpTrigger(
+            name = "getDeliveryFacilitiesBulk",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/facilities"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        try {
+            return run {
+                val body = request.body ?: return HttpUtilities.bad(request, "Body must be provided")
+                val reportIds = body.split(",").mapNotNull { ReportId.fromString(it) }
+
+                val facilities = deliveryFacade.findDeliveryFacilitiesBulk(
+                    reportIds,
+                    HistoryApiParameters(request.queryParameters).sortDir,
+                    FacilityListApiParameters(request.queryParameters).sortColumn
+                )
+
+                HttpUtilities.okResponse(
```

Comment:
I think you can maybe `HttpUtilities.okJSONResponse`?

Reply:
Some of the mapping code on tests misbehaves with it, and since there's a bunch of places still using okResponse I think it would be easier to tackle them all later